runtime/compiler/z/codegen/J9CodeGenerator.cpp

```diff
@@ -221,6 +221,24 @@ J9::Z::CodeGenerator::createLinkage(TR_LinkageConventions lc)
    return linkage;
    }
 
+/**
+ * Check if BNDS check should use a CLFI
+ */
+bool
+J9::Z::CodeGenerator::bndsChkNeedsLiteralFromPool(TR::Node *node)
```

Comment:
I don't think this API should exist. Looking at the use case, the only use case is in Dynamic Literal Pool in a path guarded by a `OMR::Node::isBndCheck`. The first child of a `TR::BNDCHK` is of type `TR::Int32` therefore the value in this function will always fall in the range of `[GE_MIN_IMMEDIATE_VAL, GE_MAX_IMMEDIATE_VAL]` and hence this API will always return false.

Can we instead deprecate the function and fold it as `false`?

Reply:
Folded the function as `false`.

Reply:
@wbh123456 the function still exists. While the single place where it is used was folded as `false` now the function has become dead and hence we can remove it entirely from the codebase. Could we do this as well?

Reply:
Sorry did not notice I put the function definition back in. I have already removed it in the new force push.